### PR TITLE
fix some of AbstractPlatform inheritors that preventing \PDO disconnect

### PR DESCRIPTION
**Pull request by knife-fighter**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

https://github.com/laminas/laminas-db/issues/5 bugfix

### CODE REVIEW

File: src/Adapter/Platform/Mysql.php [hunk 1]
```diff
@@ -58,14 +57,13 @@ public function setDriver($driver)
         if ($driver instanceof Mysqli\Mysqli
             || ($driver instanceof Pdo\Pdo && $driver->getDatabasePlatformName() == 'Mysql')
             || ($driver instanceof \mysqli)
-            || ($driver instanceof \PDO && $driver->getAttribute(\PDO::ATTR_DRIVER_NAME) == 'mysql')
```
Comment (michalbundyra) [inline src/Adapter/Platform/Mysql.php]: I am afraid we cannot remove it here, as it is BC Break.
Any chance we just keep it here but not use?

**Comment by jadrovski** [comment]
@michalbundyra  Okay, I just wrote integration test for adapter with pdo issue, so how do I push it? Should I create another branch and pull-request it? Second question is, this test is using shell_exec function to verify connection, because isConnected method is lieing in this case, can we use it in tests?
`` ``
$ testConnection = function () {
    $ mypid = getmypid ();
    $ lsof = shell_exec ("lsof -i -P -n | grep 3306 | grep $ mypid");
    return $ lsof! == null;
};
`` ``

Then, we will rewrite the fix to be sure that it will not BC break.

**Comment by michalbundyra** [comment]
@jadrovski In general it could be in this PR, but as long as you already have a lot of changes here maybe it will be better start another PR from the failing test case?

I'd like to see failing test case and then fixes which solve the issue and the test will be passing.

We already have in the library some integration tests so it can be added there. Just find a right place to put this one. I guess we can use `shell_exec`, basically tests must be failing on *nix env and then fix must fix the failing tests. Thanks!

**Comment by jadrovski** [comment]
@michalbundyra Okay, thank you for explanation. Here it is #44.
Make sure to run tests against tcp connection of mysql, socket connection (on 'localhost' value) will pass.

**Comment by michalbundyra** [comment]
Closing in favour of #44.

We would need to deprecate usage of \PDO in Platforms, but this will be for the next minor.

Thanks @knife-fighter!

**Pull request closed without merging**